src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2596,10 +2604,21 @@ namespace winrt::TerminalApp::implementation
     }
     void TerminalPage::WindowName(const winrt::hstring& value)
     {
+        const bool oldIsQuakeMode = IsQuakeWindow();
         if (_WindowName != value)
         {
             _WindowName = value;
             _PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"WindowNameForDisplay" });
+
+            // If we're entering quake mode, or leaving it
+            if (IsQuakeWindow() != oldIsQuakeMode)
+            {
+                // If we're entering QM from ~FM, then this will enter FM
+                // If we're entering QM from FM, then this will do nothing
+                // If we're leaving QM (we're already in FM), then this will do nothing
+                _SetFocusMode(_isInFocusMode);
```

Comment:
A continuation of my confusion from above, why not do `ToggleFocusMode`?

Reply:
meh, this one gets to be more explicit in what it's trying to do because it's already inside the TerminalPage. Technically, yes, `ToggleFocusMode` would work, by setting it to false, but then staying in focus mode anyways, but ¯\\\_(ツ)_/¯